### ISSUE TITLE
Add support for extend lifecycle methods

### ISSUE THREAD

**Issue by Muyangmin** [armandAkop/Lifecycle-Sorter#3]
Hey I'm using this nice plugin, but I met some problem.
In my app I have `BaseActivity` & `BaseFragment` class, and there are some extra template methods, their execute order is certain, e.g:  
```Java 
    @Override
    protected final void onCreate(Bundle savedInstanceState) {
        L.v(LOG_TAG, "----- onCreate -----");
        super.onCreate(savedInstanceState);
        parseIntentData(getIntent(), false);
        setContentView(getLayoutId());
        findViews();
        setListeners();
    }
```
In the snippet above, the `parseIntentData`, `getLayoutId`, `findViews` and `setListeners` methods has certain execute order due to android lifecycle and synchronized principle.
**So I wanna to sort those custom methods as I want, but the plugin not support yet**. Consider the difference between projects, could you consider add a config entry*(a dialog with method name input and sort)* for all methods?
Thank you very much.

**Comment by i-aakopian** [comment]
Can you elaborate a little further, maybe with an example of output? Not quite understanding. You'd like those custom methods (parseIntentData, getLayoutId, findViews, setListeners) to be sorted in the order they you are calling them?

**Comment by Muyangmin** [comment]
Yeah, often in the order I'm calling, but for more flexibility,  I would like to sort those customized methods as an order I can specify explicitly. Just like `GsonFormat` plugin, it would show a dialog with default methods (i.e. standard Android lifecycle method), and i can type methods name and change their order manually.*(I tried so much times but cannot upload screenshot image :sweat:  )*

Though auto detecting and sorting those methods sounds very cool, but to reduce the dev work, I think this plugin **needn't to detect those methods**; just let those who want to use this feature **input manually is OK**.
Wrote so much but still not know whether i have described clear, sorry about my pool english.

**Comment by i-aakopian** [comment]
I think I get the gist of what you're asking, and it does seem like a pretty neat idea that could be useful. I will give it some more thought through the week and see if I can come up with a prototype soon. Thanks for the feedback!

**Comment by Muyangmin** [comment]
Waiting for your good news :kissing_heart: !
